File: backend/msreserva/src/main/java/com/dac/msreserva/model/EstadoReserva.java
package com.dac.msreserva.model;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

@Entity
@Table(name = "estado_reserva", schema = "reserva_transaction")
public class EstadoReserva {
    @Id
    private Long codigo;
    private String sigla;
    private String descricao;

    public EstadoReserva() {}

    public EstadoReserva(Long codigo, String sigla, String descricao) {
        this.codigo = codigo;
        this.sigla = sigla;
        this.descricao = descricao;
    }

    // Getters and Setters
    public Long getCodigo() { return codigo; }
    public void setCodigo(Long codigo) { this.codigo = codigo; }

    public String getSigla() { return sigla; }
    public void setSigla(String sigla) { this.sigla = sigla; }

    public String getDescricao() { return descricao; }
    public void setDescricao(String descricao) { this.descricao = descricao; }
}

File: backend/reserva-service/src/main/kotlin/br/ufpr/dac/reserva_service/domain/EstadoReserva.kt
package br.ufpr.dac.reserva_service.domain

import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table

@Entity
@Table(name = "estado_reserva", schema = "emiratads_reserva_transaction")
data class EstadoReserva (
    @Id
    val codigo: Long,
    val sigla: String,
    val descricao: String
)